src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,99 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private void grow() {
+        elements = Arrays.copyOf(elements, elements.length + elements.length / 2);
+    }
+
+    private boolean isIndexValid(int index) {
+        if (index >= 0 && index < size) {
+            return true;
+        }
+        throw new ArrayIndexOutOfBoundsException("Wrong index. The index must be between"
+                + " 0 and size: " + size);
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == elements.length - 1) {
+            grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 ^ index > size) {
+            throw new ArrayIndexOutOfBoundsException("Wrong index. The index must be between"
+                    + " 0 and size: " + size);
+        }
+        if (size == elements.length - 1) {
```

Comment:
same

Reply:
I understood. Under this condition, the last element of the array is never used. Thank you)